torch/testing/_asserts.py

```diff
@@ -30,6 +31,17 @@ class UsageError(Exception):  # type: ignore[no-redef]
         pass
 
 
+# 'numpy' is not a hard requirement of 'torch'. If it is available, we can import the underlying data directly with
+# 'torch.from_numpy()'. Otherwise we treat 'numpy.ndarray's the same as any other unknown object.
+try:
+    import numpy as np
+
+    _NUMPY_AVAILABLE = True
+except ImportError:
+    np = None  # type: ignore[assignment]
+    _NUMPY_AVAILABLE = False
```

Comment:
`torch.as_tensor` is equivalent to `np.from_numpy`. I suggest replacing `from_numpy` usages with `as_tensor` to eliminate numpy references here as not needed.